have issues with the bot api

_If you have a question, ask on the [Miscord Support server](https://discord.gg/DkmTvVz) instead of _  
## Expected Behavior
![41](https://user-images.githubusercontent.com/24510243/39005293-db71673c-43cd-11e8-97b6-1c23e287a498.PNG)

## Actual Behavior


## Steps to Reproduce the Problem

  1.update the bot
  2.npm i
  3. npm update
  4. npm start
  5. and get an error for it

## Specifications

  - Platform: win 10
  - Version (`miscord -v`):3.1.4 - 3.1.5
  - Node.js version (`node -v`):8.10.0 

_Sent from my LG Aristo using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

@Bjornskjald see with put on silly for it 
>miscord@3.1.4 start C:\Users\robbi\Desktop\miscord-master
> node bin/index.js

info login Launching Miscord v3.1.4
info login Logging in...
info login Log level: info
info updateNotifier Checking for updates...
info login Logging in to Discord...
info login Logged in to Discord
ERR! error DiscordAPIError: Invalid Form Body
ERR! error name: This field is required
ERR! error     at item.request.gen.end (C:\Users\robbi\Desktop\miscord-master\node_modules\discord.js\src\client\rest\RequestHandlers\Sequential.js:71:65)
ERR! error     at then (C:\Users\robbi\Desktop\miscord-master\node_modules\snekfetch\src\index.js:215:21)
ERR! error     at <anonymous>
ERR! error     at process._tickCallback (internal/process/next_tick.js:188:7)
ERR! error  { DiscordAPIError: Invalid Form Body
ERR! error name: This field is required
ERR! error     at item.request.gen.end (C:\Users\robbi\Desktop\miscord-master\node_modules\discord.js\src\client\rest\RequestHandlers\Sequential.js:71:65)
ERR! error     at then (C:\Users\robbi\Desktop\miscord-master\node_modules\snekfetch\src\index.js:215:21)
ERR! error     at <anonymous>
ERR! error     at process._tickCallback (internal/process/next_tick.js:188:7)
ERR! error   stack: 'DiscordAPIError: Invalid Form Body\nname: This field is required\n    at item.request.gen.end (C:\\Users\\robbi\\Desktop\\miscord-master\\node_modules\\discord.js\\src\\client\\rest\\RequestHandlers\\Sequential.js:71:65)\n    at then (C:\\Users\\robbi\\Desktop\\miscord-master\\node_modules\\snekfetch\\src\\index.js:215:21)\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:188:7)',
ERR! error   name: 'DiscordAPIError',
ERR! error   message: 'Invalid Form Body\nname: This field is required',
ERR! error   path: '/api/v7/guilds/413159476879949825/channels',
ERR! error   code: 50035 }
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! miscord@3.1.4 start: `node bin/index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the miscord@3.1.4 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

@Bjornskjald here logs
Desktop\miscord-master>npm start

> miscord@3.1.4 start C:\Users\robbi\Desktop\miscord-master
> node bin/index.js

info login Launching Miscord v3.1.4
info login Logging in...
info login Log level: info
info updateNotifier Checking for updates...
info login Logging in to Discord...
info login Logged in to Discord
ERR! error DiscordAPIError: Invalid Form Body
ERR! error name: This field is required
ERR! error     at item.request.gen.end (C:\Users\robbi\Desktop\miscord-master\node_modules\discord.js\src\client\rest\RequestHandlers\Sequential.js:71:65)
ERR! error     at then (C:\Users\robbi\Desktop\miscord-master\node_modules\snekfetch\src\index.js:215:21)
ERR! error     at <anonymous>
ERR! error     at process._tickCallback (internal/process/next_tick.js:188:7)
ERR! error  { DiscordAPIError: Invalid Form Body
ERR! error name: This field is required
ERR! error     at item.request.gen.end (C:\Users\robbi\Desktop\miscord-master\node_modules\discord.js\src\client\rest\RequestHandlers\Sequential.js:71:65)
ERR! error     at then (C:\Users\robbi\Desktop\miscord-master\node_modules\snekfetch\src\index.js:215:21)
ERR! error     at <anonymous>
ERR! error     at process._tickCallback (internal/process/next_tick.js:188:7)
ERR! error   stack: 'DiscordAPIError: Invalid Form Body\nname: This field is required\n    at item.request.gen.end (C:\\Users\\robbi\\Desktop\\miscord-master\\node_modules\\discord.js\\src\\client\\rest\\RequestHandlers\\Sequential.js:71:65)\n    at then (C:\\Users\\robbi\\Desktop\\miscord-master\\node_modules\\snekfetch\\src\\index.js:215:21)\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:188:7)',
ERR! error   name: 'DiscordAPIError',
ERR! error   message: 'Invalid Form Body\nname: This field is required',
ERR! error   path: '/api/v7/guilds/413159476879949825/channels',
ERR! error   code: 50035 }
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! miscord@3.1.4 start: `node bin/index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the miscord@3.1.4 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

--- a/README.md
+++ b/README.md
@@ -82,7 +82,7 @@ Configuration file:
     <td><code>{}</code></td>
   </tr>
   <tr>
-    <td rowspan="7">
+    <td rowspan="8">
       Messenger<br />
       <code>messenger.*</code><br />
       <code>MESSENGER_*</code>
@@ -134,6 +134,13 @@ Configuration file:
     <td>Messenger chat filtering (see <a href="../../wiki/filtering">here</a>)</td>
     <td>:heavy_check_mark:</td>
     <td><code>[]</code></td>
+  </tr>
+  <tr>
+    <td><code>separateImages</code></td>
+    <td><code>SEPARATE_IMAGES</code></td>
+    <td>Sends images on Messenger separate from text</td>
+    <td>:heavy_check_mark:</td>
+    <td><code>false</code></td>
   </tr>
   <tr>
     <td rowspan="6">

--- a/lib/config/getConfig.js
+++ b/lib/config/getConfig.js
@@ -73,7 +73,10 @@ function getConfig (configPath) {
 function mergeDeep (...objects) {
   const isObject = obj => obj && typeof obj === 'object' && !Array.isArray(obj)
   return objects.reduce((prev, obj) => {
-    Object.keys(obj).forEach(key => { prev[key] = isObject(prev[key]) && isObject(obj[key]) ? mergeDeep(prev[key], obj[key]) : obj[key] })
+    Object.keys(obj).forEach(key => {
+      if (!obj[key]) return
+      prev[key] = isObject(prev[key]) && isObject(obj[key]) ? mergeDeep(prev[key], obj[key]) : obj[key]
+    })
     return prev
   }, {})
 }

--- a/lib/discord/createMessage.js
+++ b/lib/discord/createMessage.js
@@ -33,7 +33,7 @@ module.exports = (thread, sender, message, showFullNames) => {
     if (attach.image) {
       opts.files = [ resolveImageURL(attach) ]
     }
-    if (attach.url.startsWith('https://l.facebook.com/l.php')) {
+    if (attach.url.startsWith('http://l.facebook.com/l.php')) {
       attach.url = url.parse(attach.url, true).query.u
     }
     return [ discordMessage.includes(attach.url) ? discordMessage : (discordMessage + '\n' + attach.url), opts ]

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "miscord",
-  "version": "3.1.5",
+  "version": "3.1.6",
   "description": "Facebook Messenger to Discord bridge",
   "main": "index.js",
   "scripts": {

